assignment7.txt

```diff
@@ -0,0 +1,14 @@
+Jack Rosen
+1. ++*p increments the value at point p. *p++ increments p and then returns the value. *++p increments p and then returns the value as well.
```

Comment:
Hmm, not quite. The expression ++*p has two operators of same precedence, so compiler looks for associativity. Associativity of operators is right to left. Therefore the expression is treated as ++(*p). The expression *p++ is treated as *(p++) as the precedence of postfix ++ is higher than *. The expression *++p has two operators of same precedence, so compiler looks for associativity. Associativity of operators is right to left. Therefore the expression is treated as *(++p).